lib/facter/util/ip.rb

```diff
@@ -80,7 +80,10 @@ def self.get_interfaces
 
   def self.get_all_interface_output
     case Facter.value(:kernel)
-    when 'Linux', 'OpenBSD', 'NetBSD', 'FreeBSD', 'Darwin', 'GNU/kFreeBSD', 'DragonFly', 'AIX'
+    when 'Linux'
+      output = Facter::Util::IP.exec_ip(["link"])
```

Comment:
Based on the discussion in FACT-184, it's likely safer to first try `ip link`, then if it's not available call `ifconfig`.